### PR TITLE
support metrics in orca.tf.estimator.from_keras and support partition in orca.data

### PR DESCRIPTION
**Pull request by yangw1234**
sample code:
```python
import numpy as np
train_data = np.load("../../data/train.npy", allow_pickle=True)
val_data = np.load("../../data/val.npy", allow_pickle=True)
import zoo.orca.data as orca_data
train_x_shards = orca_data.partition(train_data)
val_x_shards = orca_data.partition(val_data)
metrics = {}
metrics["total_loss"] = model.keras_model.total_loss

from zoo.orca.learn.tf.estimator import Estimator

estimator = Estimator.from_keras(model.keras_model, metrics=metrics, model_dir="./logs")
estimator.fit(data=train_x_shards,
              epochs=30,
              batch_size=32,
              validation_data=val_x_shards,
              hard_code_batch_size=True,
              session_config=tf.ConfigProto(inter_op_parallelism_threads=2, intra_op_parallelism_threads=24))

```

### CODE REVIEW

File: pyzoo/zoo/orca/data/shard.py [hunk 1]
```diff
@@ -372,3 +374,42 @@ def value(self):
 
     def unpersist(self):
         self.broadcast_data.unpersist()
+
+
+def partition(data):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/shard.py:379]: move this to `XShards.partition`

File: pyzoo/zoo/orca/data/shard.py [hunk 2]
```diff
@@ -60,6 +60,51 @@ def load_pickle(cls, path, minPartitions=None):
...
+        :param data: np.ndarray, a tuple, list, dict of np.ndarray, or a nested structure
+        made of tuple, list, dict with ndarray as the leaf value
+        :return: a SparkXShards
+        """
+        sc = init_nncontext()
+        node_num, core_num = get_node_and_core_number()
+        total_core_num = node_num * core_num
+        import numpy as np
+        type_err_msg = """
+The types supported in zoo.orca.data.shard are
```
Comment (jason-dai) [inline pyzoo/zoo/orca/data/shard.py:76]: zoo.orca.data.XShards

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/3975/?auto_refresh=true jenkins passed

File: pyzoo/zoo/orca/data/shard.py (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/data/shard.py]: fixed

File: pyzoo/zoo/orca/data/shard.py (hunk 1 above)
Comment (yangw1234) [reply pyzoo/zoo/orca/data/shard.py]: fixed

**Pull request merged**